lib/ibandit/swedish_details_converter.rb

```diff
@@ -0,0 +1,47 @@
+module Ibandit
+  module SwedishDetailsConverter
+    def self.convert(account_number)
+      cleaned_account_number =
+        account_number.gsub(/[-.\s]/, '').gsub(/\A0+/, '')
+
+      bank_details = bank_details_for(cleaned_account_number.slice(0, 4))
+
+      if bank_details.nil?
+        return { bank_code: nil, account_number: cleaned_account_number }
+      end
+
+      clearing_code_length = bank_details.fetch(:clearing_code_length)
+
+      clearing_code = cleaned_account_number.slice(0, clearing_code_length)
+      local_account_number = cleaned_account_number[clearing_code_length..-1]
+
+      iban_account_number =
+        if bank_details.fetch(:type) == 1
+          (clearing_code + local_account_number.rjust(7, '0')).rjust(17, '0')
+        elsif bank_details[:zerofill]
+          (clearing_code + local_account_number.rjust(10, '0')).rjust(17, '0')
+        else
+          local_account_number.rjust(17, '0')
+        end
+
+      {
+        bank_code: bank_details && bank_details[:bank_code].to_s,
```

Comment:
Don't think the `bank_details &&` check is needed, as if `bank_details` is `nil` we've already returned on line 10.